smart-home-provider/cloud/auth-provider.js

```diff
@@ -304,6 +302,53 @@ Auth.registerAuth = function(app) {
 // &grant_type=authorization_code
 
 
+/**
+ * Decode authorization into client_id and client_secret.
+ * @return {{}}
+ * {
+ *   client_id: "CLIENT_ID",
+ *   client_secret: "CLIENT_SECRET"
+ * }
+ */
+function getHeaderAuthorization(req){
+  let headerAuthor = {};
```

Comment:
Do we need to declare this at the top? It seems like the value can be returned when the value is set, and return an empty object as the default case.

Reply:
Hmm... maybe I can remove this variable entirely since I can return from the inside.

(Maybe I did that for debugging purpose? I can't remember.)